HaxmLoader/haxm_loader.c

```diff
@@ -166,74 +174,38 @@ int DriverInstall(char *sysFileName)
     return retval;
 }
 
-int WriteToReg(int mem_limit, int log_level)
-{
-    HKEY hk = NULL;
-    int retval = 0;
-
-    if (RegCreateKeyEx(HKEY_LOCAL_MACHINE, "SOFTWARE\\HAXM\\HAXM",
-                       0, NULL, REG_OPTION_NON_VOLATILE, KEY_WRITE,
-                       NULL, &hk, NULL) != ERROR_SUCCESS) {
-        printf("%s(): failed to create/open registry key\r\n",
-               __FUNCTION__);
-        PrintErrorMessage();
-        return 1;
-    }
-
-    if (RegSetValueEx(hk, "MemLimit", 0,
-                      REG_DWORD, (const BYTE *)&mem_limit,
-                      sizeof(DWORD)) != ERROR_SUCCESS) {
-        printf("%s(): failed to set MemLimit\r\n", __FUNCTION__);
-        PrintErrorMessage();
-        retval = 1;
-    }
-
-    if (RegSetValueEx(hk, "LogLevel", 0,
-                      REG_DWORD, (const BYTE *)&log_level,
-                      sizeof(DWORD)) != ERROR_SUCCESS) {
-        printf("%s(): failed to set LogLevel\r\n", __FUNCTION__);
-        PrintErrorMessage();
-        retval = 1;
-    }
-
-    if (retval == 0)
-        printf("%s(): write mem_limit (%dMB) and log level (%d) to "
-               "registry done\r\n", __FUNCTION__, mem_limit, log_level);
-    return retval;
-}
-
-void PrintUsage(void)
+static void PrintUsage(void)
 {
     printf("Usage: HaxmLoader [mode]\r\n");
     printf("  Modes:\r\n");
     printf("    -i <filename>: install driver (*.sys, "
            "must be in current dir and signed)\r\n");
     printf("    -u: uninstall driver\r\n");
-    printf("    -s <mem_limit> <log_level>: set memory "
-           "limit (in MB) and log level (0~3)\r\n");
 }
 
-int __cdecl main(int argc, char *argv[])
+int __cdecl wmain(int argc, wchar_t *argv[])
 {
+    wchar_t *modeStr;
+    size_t modeStrLen = 0;
+
     if (argc == 1) {
         PrintUsage();
         return 0;
     }
-    if (argv[1][0] == '-') {
-        switch (argv[1][1]) {
-            case 'i':
+
+    modeStr = argv[1];
+    // Expect modeStr == { L'-', modeChar, NULL }
+    if (StringCchLengthW(modeStr, 3, &modeStrLen) == S_OK &&
+        modeStrLen == 2 && modeStr[0] == L'-') {
+        switch (modeStr[1]) {
+            case L'i':
                 if (argc == 3)
                     return DriverInstall(argv[2]);
                 break;
-            case 'u':
+            case L'u':
                 if (argc == 2)
                     return DriverUninstall();
                 break;
-            case 's':
-                if (argc == 4)
-                    return WriteToReg(atoi(argv[2]),
-                                      atoi(argv[3]));
-                break;
             default:
                 break;
```

Comment:
Adding a PrintUsage here?

Reply:
Yes, in line 244 which is out of this display, PrintUsage is called in "default" case. :)